cucumber-tsflow/src/binding-decorator.ts

```diff
@@ -177,9 +177,9 @@ const ensureSystemBindings = _.once(() => {
 
     scenarioContext.addExternalObject(scenarioInfo);
     scenarioContext.addExternalObject(new WorldParameters(this.parameters));
-    scenarioContext.addExternalObject(new CucumberLog(this.log.bind(this)));
+    scenarioContext.addExternalObject(new CucumberLog(this.log?.bind(this)));
     scenarioContext.addExternalObject(
-      new CucumberAttachments(this.attach.bind(this))
+      new CucumberAttachments(this.attach?.bind(this))
```

Comment:
The optional chaining would instantiate the classes with an invalid state. They should be checked and instantiated only if the functions exist:

```ts
if (this.log) scenarioContext.addExternalObject(new CucumberLog(this.log.bind(this)));
```

And the same for attachments